drivers/include/drv8825.hpp

```diff
@@ -19,27 +20,23 @@ class drv8825 : public hal::servo {
         hal::output_pin& m_direction_pin;
         hal::output_pin& m_step_pin;
         hal::steady_clock& m_clock;
-        std::array<hal::output_pin&,3> m_mode_pins;
+        hal::time_duration m_step_half_period;
+        std::array<hal::output_pin*,3> m_mode_pins;
+
         long m_partial_steps=0;// total 1/32 steps taken from pos 0 (so reverse directions steps subtract)
         double m_partial_steps_to_deg=1;
         step_factor m_step_factor = step_factor::one;
+
+    public:
```

Comment:
Having multiple public and private sections can be a bit confusing. You can limit this to just 1 private and one public by moving the private members to the bottom and combining the public sections.